pkg/pkg.bzl

```diff
@@ -140,7 +153,19 @@ def _pkg_tar_impl(ctx):
         },
         use_default_shell_env = True,
     )
-    return OutputGroupInfo(out = [ctx.outputs.out])
+    return [
+        # WHY a seperate OutputGroup?
+        # OutputGroupInfo(out = [output_file]),
+        DefaultInfo(
+            files = depset([output_file]),
+            # This is sort of ugly that the object itself must be runfiles.
+            runfiles = ctx.runfiles(files = [output_file]),
+        ),
+        PackageArtifactInfo(
+            label = ctx.label.name,
+            file_name = output_name,
+        )
+    ]
```

Comment:
Yes. The rules using naming rewrites to get a file name different from the target name would also have to produce a manifest of targets to actual names.
I don't really like having to go through a manifest, but bazel crossed that bridge with windows support. Since symlinks are not always great there, we do all runfiles through a manifest, so that we don't have to build a physical runfiles tree.

Reply:
I've done 2 things to address this.

1. If you say
`pkg_deb(name="foo", package_file="libfoo-{cpu}.deb")`
You will get a two outputs.
- the *file* libfoo-k8.deb
- and a *symlink* foo.deb pointing to libfoo-k8.deb

2. The pkg_deb rule returns a PackageArtifactInfo provider that give the label ":foo.deb", and the actual filename.

This allows you to build repackagers that only know the Bazel label name, but need to find the real filename.  And, most rules consuming the output simply work by consuming the label.

Reply:
Going back to your first comment, before you changed the subject....
Yes, we could use the same technique in pkg_filegroup.  That would allow you to take a set of files and say they should virtually appear in a folder based on the `--cpu` we are targeting.
I think that is totally compatible with what I have started here. I'm just not going to try it today because we are still iterating on the pkg_filegroup design.